ext/standard/basic_functions.stub.php

```diff
@@ -596,3 +596,231 @@ function substr_compare(string $main_str, string $str, int $offset, int $length
+
+/* math.c */
+
+/**
+ * @param int|float $number
+ * @return int|float
+ */
+function abs($number) {}
+
+function ceil(float $number): float {}
+
+function floor(float $number): float {}
+
+/** @return float|false */
+function round($number, int $precision = 0, int $mode = PHP_ROUND_HALF_UP) {}
+
+function sin(float $number): float {}
+
+function cos(float $number): float {}
+
+function tan(float $number): float {}
+
+function asin(float $number): float {}
+
+function acos(float $number): float {}
+
+function atan(float $number): float {}
+
+function atanh(float $number): float {}
+
+function atan2(float $y, float $x): float {}
+
+function sinh(float $number): float {}
+
+function cosh(float $number): float {}
+
+function tanh(float $number): float {}
+
+function asinh(float $number): float {}
+
+function acosh(float $number): float {}
+
+function expm1(float $number): float {}
+
+function log1p(float $number): float {}
+
+function pi(): float {}
+
+function is_finite(float $number): bool {}
+
+function is_nan(float $number): bool {}
+
+function intdiv(int $dividend, int $divisor): int {}
+
+function is_infinite(float $number): bool {}
+
+/** @return int|float */
+function pow($base, $exp) {}
+
+function exp(float $number): float {}
+
+/** @return float|false */
+function log(float $number, float $base = UNKNOWN) {}
+
+function log10(float $number): float {}
+
+function sqrt(float $number): float {}
+
+function hypot(float $x, float $y): float {}
+
+function deg2rad(float $number): float {}
+
+function rad2deg(float $number): float {}
+
+/** @return int|float */
+function bindec(string $binary_string) {}
+
+/** @return int|float */
+function hexdec(string $hex_string) {}
+
+/** @return int|float */
+function octdec(string $octal_string) {}
+
+function decbin($number): string {}
+
+function decoct($number): string {}
+
+function dechex($number): string {}
+
+/** @return string|false */
+function base_convert($number, int $frombase, int $tobase) {}
+
+function number_format(float $number, int $decimals = 0, string $decimal_point = "." , string $thousands_separator = ","): string {}
+
+function fmod(float $x, float $y): float {}
+
+function fdiv(float $dividend, float $divisor): float {}
+
+/* microtime.c */
+
+#ifdef HAVE_GETTIMEOFDAY
+/** @return string|float */
+function microtime(bool $get_as_float = false) {}
+
+/** @return array|float */
+function gettimeofday(bool $return_float = false) {}
+#endif
+
+#ifdef HAVE_GETRUSAGE
+/** @return array|false */
+function getrusage(int $who = 0) {}
+#endif
+
+/* pack.c */
+
+/** @return string|false */
+function pack(string $format, ...$args) {}
+
+/** @return array|false */
+function unpack(string $format, string $data, int $offset = 0) {}
+
+/* password.c */
+
+function password_get_info(string $hash): ?array {}
+
+/** @return string|false */
+function password_hash(string $password, $algo, $options = UNKNOWN) {}
```

Comment:
```suggestion
function password_hash(string $password, $algo, array $options = []) {}
```

I've removed the ability to pass an object here in https://github.com/php/php-src/commit/63a20cb400d064bc56254896dd8e1e921df27af8. Pretty sure that someone mixed up a zpp modifier back then, and this was not supposed to work.